src/test/unit/sonar_unittest.cc

```diff
@@ -102,9 +102,11 @@ TEST(SonarUnittest, TestAltitude)
     // distance 400, 22.4 degrees of roll, this corresponds to HC-SR04 effective max detection angle
     EXPECT_EQ(sonarCalculateAltitude(400, cosf(DECIDEGREES_TO_RADIANS(224))), 369);
     EXPECT_EQ(sonarGetLatestAltitude(), 369);
-    // max range, max tilt
-    EXPECT_EQ(sonarCalculateAltitude(sonarMaxRangeCm, cosf(DECIDEGREES_TO_RADIANS(sonarMaxTiltDeciDegrees))), sonarMaxAltWithTiltCm);
-    EXPECT_EQ(sonarGetLatestAltitude(), sonarMaxAltWithTiltCm);
+    // Check limits at max range max tilt, need to deal with rounding errors in cosf calculation
+    EXPECT_GE(sonarCalculateAltitude(sonarMaxRangeCm, cosf(DECIDEGREES_TO_RADIANS(sonarMaxTiltDeciDegrees - 1))), sonarMaxAltWithTiltCm - 1);
```

Comment:
there is `EXPECT_NEAR(val1, val2, abs_error);` macro..

https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md